controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -218,11 +233,82 @@ func (r *StorageConsumerReconciler) setCephResourceStatus(name string, kind stri
 	)
 }
 
+func (r *StorageConsumerReconciler) deletionPhase() error {
+	lastHeartBeat := r.storageConsumer.Status.LastHeartbeat
+	if !lastHeartBeat.IsZero() && time.Since(lastHeartBeat.Time) >= time.Duration(5*time.Minute) {
+		if err := r.annotateCephResources(); err != nil {
+			return err
+		}
+	}
+
+	if controllerutil.RemoveFinalizer(r.storageConsumer, storageConsumerFinalizer) {
+		if err := r.update(r.storageConsumer); err != nil {
+			return fmt.Errorf("failed to remove finalizer: %v", err)
+		}
+	}
```

Comment:
should we wait till the rook CR's are deleted before removing the finalizers and deleting the consumer CR?

Reply:
reimplemented a straight forward way, pls re-review. thanks.

Reply:
resolving after confirming offline.